eureka_ml_insights/models/models.py

```diff
@@ -1030,6 +1030,92 @@ def _generate(self, text_prompt, query_images=None, system_message=None):
         self.response_time = end_time - start_time
 
 
+@dataclass
+class vLLMModel(Model):
+    """This class is used to run a self-hosted language model via vLLM apis."""
+
+    model_name: str = None
+    trust_remote_code: bool = False
+    tensor_parallel_size: int = 1
+    dtype: str = "auto"
+    quantization: str = None
+    seed: int = 0
+    gpu_memory_utilization: float = 0.9
+    cpu_offload_gb: float = 0
+
+    temperature: float = 0.001
+    top_p: float = 0.95
+    top_k: int = -1
+    max_tokens: int = 2000
+    apply_model_template: bool = True
+
+    def __post_init__(self):
+        # vLLM automatically picks an available devices when get_model() is called
+        self.get_model()
+
+    def get_model(self):
+        from vllm import LLM
```

Comment:
Let's update the requirements file with whatever this package needs. As long as our tests pass after installing eureka with the new requirements file, we should be good to go.

We have pip installation and conda installation requirements and setup files.